src/main/java/com/minecolonies/core/tileentities/TileEntityColonyBuilding.java

```diff
@@ -444,7 +446,7 @@ public void saveAdditional(@NotNull final CompoundTag compound, @NotNull final H
     @Override
     public void tick()
     {
-        if (combinedInv != null)
+        if (this.building instanceof AbstractBuildingContainer container && container.pollContainerListChanged())
         {
             invalidateCapabilities();
             combinedInv = null;
```

Comment:
I'd immediately recreate the inventory right here, right after setting the inv to null, using a small utility method to return the inv if not null, else recreate it.